<a href="https://colab.research.google.com/github/MaxTechniche/DS-Unit-1-Sprint-2-Statistics/blob/master/module3/Jacob_Maxfield_LS_DS_123_Confidence_Intervals_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Confidence Intervals

Soft drinks like Coke and Pepsi are manufactured to have a standard caffeine content. For example, a 12-oz serving of Coke has 34mg of caffeine, and a 12-oz serving of Pepsi has 37.6mg of caffeine. However, fountain soft drinks are typically mixed in individual restaurant dispensers, so it is more difficult to maintain a standard level of caffeine per serving. In this study, researchers randomly sampled Coke, Diet Coke, Pepsi, and Diet Pepsi at a set of franchise restaurants and measured the caffeine content in 12oz of each soft drink2. The data is found in the Soda.xlsx dataset.

Because individuals can be sensitive to caffeine – and because the manufacturers are interested in product consistency – we wish to estimate the mean caffeine content in 12oz of Coke served in franchise restaurants using a 95% confidence interval. 

You can find the Coke data here: 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv'

The first variable is the sample ID and the second variable is the caffeine content in the 12-oz sample measured in mg.

Source: A.N. Garand and L.N. Bell (1997). "Caffeine Content of Fountain and Private-Label Store Brand Carbonated Beverages," Journal of the American Dietetic Association, Vol. 97, #2, pp. 179-182.




###1) Load the dataset and print the first few rows

In [1]:
#Import statements
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
df = pd.read_csv('https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Coke.csv')
df.head()
#Print the top 5 rows

Drink  Caffeine
0      1     47.32
1      2     43.78
2      3     48.12
3      4     43.25
4      5     46.42

###2) Calculate the mean, SD, SE caffeiene content and n for the sample.

In [3]:
df['Caffeine'].describe()

count    50.000000
mean     37.940200
std       5.243757
min      28.430000
25%      33.662500
50%      37.485000
75%      41.737500
max      48.120000
Name: Caffeine, dtype: float64

In [4]:
#Calculations here
drink_count, drink_mean, drink_std, *__ = df['Caffeine'].describe()
drink_se = drink_std/np.sqrt(drink_count)
drink_se

0.7415792024250598

###3) Find t* for a 95% confidence interval.  

Use the starter code below and fill in the degrees of freedom.

In [5]:
from scipy.stats import t

#Don't worry too much about where the 0.975 comes from.  It has to do
#with wanting to determine the *middle* 95% of the t-distribution
#We're going to learn
#how to calculate a 95% CI this easy way in just a minute.

#Recall that n = 223 for the body temp problem.
t_star = t.ppf(0.975,df=drink_count-1)
print('t_star =', t_star)

t_star = 2.009575234489209


###4) Calculate the margin of error for a 95% confidence interval for the mean caffeine content in a 12-oz Coke.



In [6]:
#Calculations here
t_margin = t_star * drink_se
t_margin

1.49025919960566

95% confidence the population mean is between the average - 1.49 and average + 1.49

###5) Calculate a 95% CI for the mean caffeine content in a 12-oz fountain Coke with the CI formula using the summary statistics and t* that you calculated above.

In [8]:
#Calculations here
print(f'Low: {drink_mean-t_margin}, High: {drink_mean+t_margin}')

Low: 36.44994080039434, High: 39.43045919960566


###6) Calculate a 95% CI for the mean caffeiene content in a 12-oz fountain Coke using the t-interval function in Python.

In [21]:
#Calculations here
t.interval(.95, drink_count-1, drink_mean, drink_se)

(36.44994080039434, 39.43045919960566)

###7) Compare the two confidence intervals you calculated.  Do they match?  Should they?

They are the same, and generally should be. There might sometimes be VERY miniscule differences depending on how you go about doing the calculations (because fo rounding).

###8) Interpret the meaning of the 95% confidence interval for the mean caffeiene content in a 12-oz fountain Coke. in a sentence or two.

We are 95% confident that the average caffine content in a 12-oz fountain coke is between 36.45 and 39.43 grams.

###9) Using the t-interval Python function, calculate a 90% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [10]:
#Calculations here
t.interval(.90, drink_count-1, drink_mean, drink_se)

(36.696904726749196, 39.1834952732508)

The estimate is more precise because we choose to narrow in on an average, but then we are less sure that the average is within that range.

###10) Using the t-interval Python function, calculate a 99% confidence interval for the mean caffeine content in a 12-oz Coke.  Is this estimate more accurate or more precise (pick one) than the 95% confidence interval?



In [12]:
#Calculations here
t.interval(.99, drink_count-1, drink_mean, drink_se)

(35.95280335285685, 39.92759664714315)

The estimate is more accurate because we choose to widen our estimated range, making the estimate less precise but more likely to hold our average.

## Stretch goals:

###1) The correspondence between confidence intervals and hypothesis tests.

Read [this](https://https://blog.minitab.com/blog/adventures-in-statistics-2/understanding-hypothesis-tests-confidence-intervals-and-confidence-levels#:~:text=If%20a%20hypothesis%20test%20produces,corresponding%20confidence%20level%20is%2095%25.&text=If%20the%20confidence%20interval%20does,the%20results%20are%20statistically%20significant.) article about the correspondence between confidence intervals and hypothesis tests.  Feel free to read the whole article, but the relevant part can be found under the heading Why P Values and Confidence Intervals Always Agree About Statistical Significance.

Imagine you work for quality control at Coke and are tasked with making sure that the caffeiene content in the fountain beverages served in restaurants is the same as in a 12-oz can of Coke (34mg).  If you believe that the mean caffeiene content in fountain coke is not 34mg, you must re-train the franchise managers to make sure the Coke served has the correct caffeiene level.

Based on the confidence interval you calculated in the assignment, do you believe that the mean caffeiene content is statistically significantly different from 34 mg in a 12-oz serving?


In [13]:
stats.ttest_1samp(df['Caffeine'], 34)

Ttest_1sampResult(statistic=5.313255802097796, pvalue=2.620466428255138e-06)

Based on our confidence interval and ammount of caffine we should expect to be in fountatin drinks, we can say that the the caffine content is statistically significantly different.

###2) If we increased the sample size from 50 to 100 but the sample mean and SD remained the same, describe **two** ways the margin of error would change.  Would the margin of error become smaller or larger?

The margin of error would become smaller.

In [15]:
t_star

2.009575234489209

In [17]:
t_star_100 = t.ppf(0.975,df=99)
t_star_100

1.9842169515086827

In [18]:
t_star-t_star_100

0.025358282980526115

If we increase the sample size to 100, but everything were to stay the same, in would improve our precision by .025 grams in either direction if we have a confidence of 95%